ui/src/wizard/ResourceGroupCreateStep.ts

```diff
@@ -38,6 +38,16 @@ export class ResourceGroupCreateStep<T extends types.IResourceGroupWizardContext
             if (wizardContext.suppress403Handling || parseError(error).errorType !== '403') {
                 throw error;
             } else {
+                const subClient = await createSubscriptionsClient(wizardContext);
+                const sub = await subClient.subscriptions.get(wizardContext.subscriptionId);
+                // if this is a Sandbox account, pick the only rg available for the user
+                if (sub.subscriptionPolicies?.quotaId && /sponsored/i.test(sub.subscriptionPolicies?.quotaId)) {
```

Comment:
Maybe we can check two things to confirm it's a sandbox subscription: "sponsored..." in quotaId and "learn-..." in subscriptionId?

Reply:
> Or maybe, If they cannot create a rg and have only one rg, we will automatically select that? This is instead of matching "sponsored" in the quotaId.

I think this would be cleaner and I can't think of any downsides. It makes sense that if the rg creation fails and there's a single existing rg, that group is probably from the concierge subscription.

Reply:
I don't think we can assume concierge if there's only one rg. Just from my own personal experience, I've had people share a single resource group from their subscription with me to test something out or troubleshoot an issue. They don't want to give me permissions to their whole subscription, but a single group is easy to do.

The way I see it, there's two ways we could get it wrong:
1. We think it's a learn subscriptions, but it's not. We skip the prompt and auto select a random resource group
2. We think it's _not_ a learn subscription, but it is. We prompt them to pick a resource group

IMO, 1 is a lot worse than 2 - so I'd rather we be extra careful to make sure it's a learn subscription. With 1, you essentially get rid of a confirmation message. For example, I may have accidentally selected the wrong subscription in the first place and the message gives me a chance to cancel and start over with the right one. With 2, it's just a few extra clicks.

Is it fair to say we don't know much about "quotaId" and "Sponsored"? If so, my suggestion would actually be to check that both `subscriptionDisplayName` equals "Concierge Subscription" and the resource group starts with "learn-". I normally don't like checking against english display names like that, but as mentioned the downside of 2 just doesn't seem that bad to me.

Reply:
I did more digging since I think that this is a valid concern.  If you look at the portal, all Concierge Subscriptions are considered to be part of the Azure Sponsorship offer.  

![image](https://user-images.githubusercontent.com/5290572/116945414-6b14dd00-ac2c-11eb-9c1a-1818d2e3ada6.png)

This offer matches the quotaId that I'm using to validate if the subscription is concierge or not

![image](https://user-images.githubusercontent.com/5290572/116945476-9b5c7b80-ac2c-11eb-9454-246a022123f2.png) (source: https://docs.microsoft.com/en-us/azure/cost-management-billing/costs/understand-cost-mgt-data)

Unfortunately, I haven't seen much documentation for quotaId in the REST API beyond [this](https://docs.microsoft.com/en-us/rest/api/resources/subscriptions/list?ranMID=24542&ranEAID=je6NUbpObpQ&ranSiteID=je6NUbpObpQ-jioMT00WFz1YxN3WOutymw&epi=je6NUbpObpQ-jioMT00WFz1YxN3WOutymw&irgwc=1&OCID=AID2000142_aff_7593_1243925&tduid=%28ir__u2o2gsvb3okftmp1kk0sohzibn2xu1oogcgx1wq900%29%287593%29%281243925%29%28je6NUbpObpQ-jioMT00WFz1YxN3WOutymw%29%28%29&irclickid=_u2o2gsvb3okftmp1kk0sohzibn2xu1oogcgx1wq900#subscriptionpolicies).

Based off the offer code, I do believe that there are other instances of Azure Sponsorship subscriptions, but I personally think that the risk is somewhat low since for any sponsored subscription, I think we would want to automatically select the rg for the customer if there were only one and they could not create more.

Reply:
I just don't think we want to auto-magically do something for the user on a subscription we don't fully understand (the "other instances of Azure Sponsorship subscriptions"). I think keeping the confirmation message in there is much safer when we're dealing with things like permissions/billing and there's little downside of the few extra clicks. Also this code affects every single extension, so another reason to be careful.

Reply:
> I propose that , ideally, the extension should detect if the subscription is a concierge/sandbox subscription that it should use the existing resource group, saving the learner from having to press the "select existing" button and then select it

I think I agree with John Papa here.  Having the creation experience hiccup for new users going through the Learn module can be pretty distasteful/off-putting especially since all the tutorials don't account for the step erroring out.

While the workaround for the user is pretty simple, picking an existing resource group for when we fail to create a new one for `sponsored` subscription policies feels relatively low risk to me.  

By using an existing resource group, I don't believe we would be changing any type of permission.  Also note that this _only_ happens if there is only 1 resource group available to the user so it's not like we are blocking them from selecting a different rg.  The only downside I can think of is that their bill might increase for creating another web app, but that was kind of the point of them pushing the + button anyway.

And it would solve every sandbox account problem 😊

Reply:
Lol tryin to throw the Paps at me. We agree on the behavior, just not how to "detect if the subscription is a concierge/sandbox". Can you guarantee that "Sponsored" equals "concierge/sandbox"? If not, my suggestion was you check for "concierge" in the subscription name and "learn" in the rg name, which should be safer and will still catch 99% of sandbox subscriptions. I don't see John Papa say "Sponsored" anywhere in your quote, but he does say "concierge" and "learner"

😊